cmd/ebpf/xdp/n3n6_entrypoint.c

```diff
@@ -158,16 +188,84 @@ static __always_inline enum xdp_action handle_gtp_packet(struct packet_context *
         return DEFAULT_XDP_ACTION;
     }
 
+    __u32 far_id = pdr->far_id;
+    __u32 qer_id = pdr->qer_id;
+    __u8 outer_header_removal = pdr->outer_header_removal;
+    
+    if (pdr->has_sdf) {
+        struct packet_context inner_context = {
+            .data = (char *)(long)ctx->data,
+            .data_end = (const char *)(long)ctx->data_end,
+        };
+
+        if (inner_context.data + 1 > inner_context.data_end)
+            return -1;
```

Comment:
It seems that -1 shouldn't be returned in this function. It is more likely to be some meaningful enum value, like DEFAULT_XDP_ACTION or XDP_DROP. Probably DEFAULT_XDP_ACTION, since the same check is performed on lines 207-211 (calling function parse_ip4) and if it fails, return code will be DEFAULT_XDP_ACTION.

Reply:
Must return DEFAULT_XDP_ACTION. Absolutelly right